hyperspy/signal_tools.py

```diff
@@ -873,10 +873,12 @@ def __init__(self, signal, navigation_mask=None, signal_mask=None):
                 self.noise_type = "shot noise"
         else:
             self.noise_type = "shot noise"
+        self.find()
```

Comment:
Why calling ``find`` here?

Reply:
I am to going it to revert it because it may not be desirable if the threshold value is too high in case of large dataset since it can take too long to perform... The idea is to save the user a click on the "find" button at the start of the spikes_removal_tool, but it may be a wrong good idea! Is it what you are concerned about?